examples/tests/client-tests.js

```diff
@@ -0,0 +1,182 @@
+/* global QUnit */
+
+'use strict';
+
+const BASE_MAIN_CHILDREN_NUMBER = 3;
+
+QUnit.module('Pictures');
+
+QUnit.test(
+  'putPicturesInPage',
+  (assert) => {
+    assert.ok(
+      typeof window.forTesting.putPicturesInPage === 'function',
+      'Create a function `putPicturesInPage` receiving an array of pictures. The function will add each picture and its attributes to a different section with the class `picture`.',
+    );
+
+    const elM = document.querySelector('main');
+    assert.strictEqual(
+      elM.childElementCount,
+      (0+BASE_MAIN_CHILDREN_NUMBER),
+      'Before using the function, the element `main` must be empty of pictures.',
+    );
+
+    // with an empty array
+    window.forTesting.putPicturesInPage([]);
+    const el = document.querySelectorAll('main section');
+
+    assert.strictEqual(
+      el.length,
+      (1),
+      'When there is no picture to be added, there must be only one `section` child to main.',
+    );
+
+    // with an array of one picture
+    let data = [{
+      id: 1,
+      file: 'http://placebear.com/400/200',
+      title: 'A bear',
+    }];
+    window.forTesting.putPicturesInPage(data);
+
+    assert.strictEqual(
+      elM.childElementCount,
+      (1+BASE_MAIN_CHILDREN_NUMBER),
+      'After passing one image entity to the function, there must be only one `section`.',
+    );
+
+    let listOfSections = elM.querySelectorAll('section.picture');
+    let section = listOfSections[0];
+    let paragraphs = section.querySelectorAll('p');
+    let as = section.querySelectorAll('a');
+    let images = as[0].querySelectorAll('img');
+    let deletes = section.querySelectorAll('div');
+
+    assert.strictEqual(
+      listOfSections.length,
+      1,
+      'There must be one section in `main`.',
+    );
+    assert.ok(
+      section.classList.contains('picture'),
+      'A `section` must have the class `picture`.',
+    );
+
+    assert.strictEqual(
+      paragraphs.length,
+      2,
+      'There must be two `<p>` elements in a `section`.',
```

Comment:
this fails - is this because you're testing with author? 

I'm finishing my review of this file here because I've run out of time, sorry.